guvnor-project/guvnor-project-api/src/main/java/org/guvnor/common/services/project/service/ProjectSearchService.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.guvnor.common.services.project.service;
+
+import java.util.Collection;
+
+import org.guvnor.common.services.project.model.Project;
+import org.jboss.errai.bus.server.annotations.Remote;
+
+/**
+ * {@link Project} remote search interface
+ */
+@Remote
+public interface ProjectSearchService {
```

Comment:
still not completely confident about this search service; search for global projects sounds like too much (think abotu hundreds of projects in a large organization)... gut feeling is that this search would need some context.. and with a context would be possible to load everything into a data structure and send it to client to handle